src/vtt-segment-loader.js

```diff
@@ -477,13 +477,17 @@ export default class VTTSegmentLoader extends SegmentLoader {
       return;
     }
 
-    const timestampmap = segmentInfo.timestampmap;
-    const diff = (timestampmap.MPEGTS / ONE_SECOND_IN_TS) - timestampmap.LOCAL + mappingObj.mapping;
+    const { MPEGTS, LOCAL } = segmentInfo.timestampmap;
+    const diff = (MPEGTS / ONE_SECOND_IN_TS) - LOCAL + mappingObj.mapping;
 
     segmentInfo.cues.forEach((cue) => {
-      // First convert cue time to TS time using the timestamp-map provided within the vtt
-      cue.startTime += diff;
-      cue.endTime += diff;
+      const duration = cue.endTime - cue.startTime;
+      const startTime = this.sourceType_ === 'dash' ?
```

Comment:
For my own understanding, why doesn't DASH need to handle rollover?

Reply:
Actually, this is an interesting question... 
MPEGTS has a limitation for timestamps as 2^33. If the timestamp is more than 2^33, it will roll over to 0. It will continue to increase monotonically until it reaches 2^33 and rolls over once again, etc...

Initially, I decided to keep it isolated to HLS, but technically, it is possible to use MPEGTS with a DASH (I haven't seen it, though).
So I would assume that it might be better to check that MPEGTS timestamp map > 0 rather than isolate it to HLS only

Reply:
I updated this check